src/app.js

```diff
@@ -1,5 +1,155 @@
 const express = require('express');
+const { sayHello, uppercase, lowercase, firstCharacters } = require('./lib/strings');
+const { add, subtract, multiply, divide, remainder, quotient } = require('./lib/numbers');
+const { negate, truthiness, isOdd, startsWith, containsVowels } = require('./lib/booleans');
+const {
+  getNthElement,
+  arrayToCSVString,
+  addToArray2,
+  elementsStartingWithAVowel,
+  removeNthElement2,
+} = require('./lib/arrays');
 
 const app = express();
 
+app.use(express.json());
+
+app.get('/', (req, res) => {
+  res.send('Hello, world!');
+});
+
+// Strings
+
+app.get('/strings/hello/:str', (req, res) => {
+  const string = req.params.str;
+  res.status(200).json({ result: sayHello(string) });
+});
+
+app.get('/strings/upper/:str', (req, res) => {
+  const string = req.params.str;
+  res.status(200).json({ result: uppercase(string) });
+});
+
+app.get('/strings/lower/:str', (req, res) => {
+  const string = req.params.str;
+  res.status(200).json({ result: lowercase(string) });
+});
+
+app.get('/strings/first-characters/:str', (req, res) => {
+  const string = req.params.str;
+  const i = req.query.length;
+  const n = i > 0 ? i : 1;
+  res.status(200).json({ result: firstCharacters(string, n) });
+});
+
+// Numbers
+
+app.get('/numbers/add/:num1/and/:num2', (req, res) => {
+  const number1 = parseInt(req.params.num1);
+  const number2 = parseInt(req.params.num2);
+  return Number.isNaN(number1) || Number.isNaN(number2)
+    ? res.status(400).json({ error: 'Parameters must be valid numbers.' })
+    : res.status(200).json({ result: add(number1, number2) });
+});
+
+app.get('/numbers/subtract/:num1/from/:num2', (req, res) => {
+  const number1 = parseInt(req.params.num1);
+  const number2 = parseInt(req.params.num2);
+  return Number.isNaN(number1) || Number.isNaN(number2)
+    ? res.status(400).json({ error: 'Parameters must be valid numbers.' })
+    : res.status(200).json({ result: subtract(number2, number1) });
+});
+
+app.post('/numbers/multiply', (req, res) => {
+  if (typeof req.body.a === 'undefined' || typeof req.body.b === 'undefined') {
+    res.status(400).json({ error: 'Parameters "a" and "b" are required.' });
+  } else {
+    const answer =
+      isNaN(req.body.a) || isNaN(req.body.b)
+        ? res.status(400).json({ error: 'Parameters "a" and "b" must be valid numbers.' })
+        : res.status(200).json({ result: multiply(req.body.a, req.body.b) });
+    return answer;
+  }
```

Comment:
Okay, so theres a few things to look at here. While this works, it uses the global `isNaN()` method which is being replaced by the more robust `Number.isNan` (see [MDN docs on it here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN))

When replacing the `isNaN` function with `Number.isNaN()`, this breaks. This is because you are only checking for whether or not the values passed in are undefined or not, then only whether they are numbers or not. When passing in strings this will break it, so an extra check has to be added in to convert the values to numbers if they are strings using `parseInt()`. 
Since `parseInt` will return `NaN` if passed a string, we can then check use `Number.isNan()` to check if it is actually a number passed in in the first place. Putting all that together, you could do something like this:
```
  let { a, b } = req.body;

  if (a === undefined || b === undefined) {
    res.status(400).json({ error: 'Parameters "a" and "b" are required.' });
  }

  if (typeof a === 'string' || typeof b === 'string') {
    a = parseInt(a, 10);
    b = parseInt(b, 10);
  }

  if (Number.isNaN(a) || Number.isNaN(b)) {
    res.status(400).json({ error: 'Parameters "a" and "b" must be valid numbers.' });
  }

  res.status(200).json({ result: multiply(a, b) });
```